src/main/java/com/amogoscode/groupe/ebankingsuite/User/UserController.java

```diff
@@ -0,0 +1,56 @@
+package com.amogoscode.groupe.ebankingsuite.User;
+
+import com.amogoscode.groupe.ebankingsuite.User.resquests.UserAuthenticationRequests;
+import com.amogoscode.groupe.ebankingsuite.User.resquests.UserRegistrationRequest;
+import com.amogoscode.groupe.ebankingsuite.exception.InvalidAuthenticationException;
+import com.amogoscode.groupe.ebankingsuite.exception.ResourceNotFoundException;
+import com.amogoscode.groupe.ebankingsuite.universal.ApiResponse;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/v1/users")
+public class UserController {
+
+    private final UserService userService;
+
+    public UserController(UserService userService) {
```

Comment:
Thank you for the comment omar. However, if we do not inject the UserService via the constructor, we will have issues passing a mock of the UserService to the UserController in the UserController test class.

Reply:
@smallingla 
what I meant is using @Autowired up of the ctor like this
![image](https://user-images.githubusercontent.com/59146036/222223618-c7abbf78-d812-4e14-b602-e3d242ced336.png)


why : 
In Spring, the @Autowired annotation is used to inject dependencies into a class. However, having multiple constructors annotated with @Autowired can lead to ambiguity and make it difficult for Spring to determine which constructor to use for dependency injection.

so to prevent others adding ctors, and ctors with/without @Autowired, best practice to add it.

Reply:
well noted omar. Thank you very much, this will be implemented in the subsequent commit

Reply:
@smallingla 
If you need someting, 
code review
or something to help with coding/Java

you can contact me in github (for free of course)

Reply:
Thank you @omarmahamid

Reply:
changes has been pushed in new commit